server.py

```diff
@@ -15,44 +13,62 @@
 
 # server class
 class Server:
-    def __init__(self, imagedata):
-        self.board = CanvasBoard(imagedata)
-        self.lock = threading.Lock()
+    def __init__(self):
+        self.__boards = {}
 
-    def update_board(self, diffs: list):
-        self.lock.acquire()
-        self.board.update_board(diffs)
-        self.lock.release()
+    def get_ids(self):
+        return self.__boards.keys()
 
+    def add_board(self, room_id, board):
+        self.__boards[room_id] = board

```

Comment:
Should we check if the room already exists here?

Reply:
Added validation